frontend/class-frontend.php

```diff
@@ -727,24 +810,23 @@ public function robots() {
 		/**
 		 * Determine $robots values for a single post
 		 *
-		 * @param    array      $robots
-		 * @param    int|string $postid The postid for which to determine the $robots values, defaults to
-		 *                              the current post
+		 * @param array      $robots
+		 * @param int|string $post_id The post ID for which to determine the $robots values, defaults to current post
 		 *
 		 * @return    array
 		 */
-		function robots_for_single_post( $robots, $postid = 0 ) {
-			if ( WPSEO_Meta::get_value( 'meta-robots-noindex', $postid ) === '1' ) {
+		function robots_for_single_post( $robots, $post_id = 0 ) {
+			if ( WPSEO_Meta::get_value( 'meta-robots-noindex', $post_id ) === '1' ) {
```

Comment:
A switch only does a loose type check, i.e. `==` not `===`. This might give quite undesirable side effects.

You can get round that by doing something along the lines of:
```php
switch( true ) {
	case ( $a === $b ) :
		//do something
		break;
}
```